md2cf/replacements.py

```diff
@@ -0,0 +1,70 @@
+import importlib.util
+import json
+import re
+from typing import List
+
+from md2cf.console_output import console
+
+
+class Replacement:
+    def __init__(
+        self, name: str, pattern: str, new_value: str, evaluate: bool = False
+    ) -> None:
+        self.name = name
+        self.pattern = pattern
+        self.new_value = new_value
+        self.evaluate = evaluate
+
+    def replace(self, page):
+        console.print(f"Performing replacement '{self.name}'")
+        if self.evaluate:
+            new_value = eval(self.new_value)
+        else:
+            new_value = self.new_value
+        page.body, count = re.subn(f"({self.pattern})", new_value, page.body)
+        console.print(f">> {count} replacements made")
+        return page
+
+    def __repr__(self) -> str:
+        return self.name
+
+
+def create_replacements(replacements, replacementfile: str) -> List[Replacement]:
+    result = []
+    commandline_replacements = (
+        [item for sublist in replacements for item in sublist] if replacements else []
+    )
+
+    # Create Replacement objects for the commandline replacements
+    for i, r in enumerate(commandline_replacements):
+        result.append(Replacement(f"CLI replacement {i}", *r.split("=", 1)))
```

Comment:
This can go wrong pretty easily if somebody forgets the `=`: it will just result in a non-obvious exception about the number of parameters when calling the constructor. I'd love to have this function receive data that's already well-structured. For example, you could parse the command line arguments in `__main.py__` (with error handling if the format is wrong) and give `create_replacement` a well-formed list of lists, e.g.

```
[["pattern", "replacement], ["another pattern", "another replacement"]]
```